alexandria/data_structures/src/tests/stack_test.cairo

```diff
@@ -57,31 +49,135 @@ fn stack_peek_test() {
+impl TDrop: Drop<T>,
+impl TCopy: Copy<T>,
+impl TPartialEq: PartialEq<T>,
+impl SDestruct: Destruct<S>>(
+    ref stack: S, val_1: T, val_2: T
+) {
+    stack.push(val_1);
+    stack.push(val_2);
+
+    let value = stack.pop();
+    match value {
+        Option::Some(result) => {
+            assert(result == val_2, 'wrong result');
+        },
+        Option::None(_) => {
+            assert(0 == 1, 'should return a value');
+        },
+    };
+
+    assert(stack.len() == 1, 'should remove item');
+}
 
+fn stack_push_pop_push_test<S,
+T,
+impl Stack: StackTrait<S, T>,
+impl TDrop: Drop<T>,
+impl TCopy: Copy<T>,
+impl TPartialEq: PartialEq<T>,
+impl SDestruct: Destruct<S>>(
+    ref stack: S, val_1: T, val_2: T, val_3: T
+) {
     stack.push(val_1);
     stack.push(val_2);
-// TODO Commented due to dangling issue
-// let value = stack.pop();
-// match value {
-//     Option::Some(result) => {
-//         assert(result == val_2, 'wrong result');
-//     },
-//     Option::None(_) => {
-//         assert(0 == 1, 'should return a value');
-//     },
-// };
+    stack.pop();
+    stack.push(val_3);
+
+    assert(stack.peek().unwrap() == val_3, 'wrong result');
+    assert(stack.len() == 2, 'should update length');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_new_test() {
+    stack_new_test(@StackTrait::<Felt252Stack, u128>::new());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_is_empty_test() {
+    stack_is_empty_test(@StackTrait::<Felt252Stack, u128>::new());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_push_test() {
+    let mut stack = StackTrait::<Felt252Stack, u128>::new();
+    stack_push_test(ref stack, 1.try_into().unwrap(), 2.try_into().unwrap());
+}
 
-// let result_len = stack.len();
-// assert(result_len == 1, 'should remove item');
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_peek_test() {
+    let mut stack = StackTrait::<Felt252Stack, u128>::new();
+    stack_peek_test(ref stack, 1.try_into().unwrap(), 2.try_into().unwrap());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_pop_test() {
+    let mut stack = StackTrait::<Felt252Stack, u128>::new();
+    stack_pop_test(ref stack, 1.try_into().unwrap(), 2.try_into().unwrap());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn felt252_stack_push_pop_push_test() {
+    let mut stack = StackTrait::<Felt252Stack, u128>::new();
+
+    stack_push_pop_push_test(
+        ref stack, 1.try_into().unwrap(), 2.try_into().unwrap(), 3.try_into().unwrap()
+    );
+}
+
+
+#[test]
+#[available_gas(2000000)]
+fn nullable_stack_new_test() {
+    stack_new_test(@StackTrait::<NullableStack, u256>::new());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn nullable_stack_is_empty_test() {
+    stack_is_empty_test(@StackTrait::<NullableStack, u256>::new());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn nullable_stack_push_test() {
+    let mut stack = StackTrait::<NullableStack, u256>::new();
+    stack_push_test(ref stack, 1.into(), 2.into());
```

Comment:
Automatically infered :) 
```suggestion
    stack_push_test(ref stack, 1, 2);
```

Reply:
which version of the compiler are you using?

Reply:
1.1. Actually now it works as you suggested.